tests/integration/targets/lookup_hashi_vault/lookup_hashi_vault/tasks/tinyproxy_server.yml

```
@@ -0,0 +1,19 @@
+---
+- name: Install tinyproxy
+  shell: |
+    apt-get update
+    DEBIAN_FRONTEND=noninteractive apt-get --assume-yes install tinyproxy
+  changed_when: false
+
+- name: Configure tinyproxy
+  copy:
+    content: |
+      Port 8001
+      MaxClients 100
+      StartServers 10
+    dest: /tinyproxy.conf
+
+- name: Start tinyproxy
+  shell: |
+    tinyproxy -c /tinyproxy.conf
+  changed_when: false
```

```suggestion
  notify: "name of cleanup handler"
```
We don't need to mess with the changed status, but as mentioned above a handler should be added that will kill the proxy at the end of the tests. These tests are configured to run handlers even when there's a failure.

---

done